File: Frontend/rostam-academy/src/app/academy/classes/class.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {Observable,of} from 'rxjs';
import {Class} from './class.model';
import {catchError} from 'rxjs/operators';

@Injectable()

export class ClassService {
  private url = "http://localhost:5000/api/classes";

  constructor(private httpClient : HttpClient){ }

  getAll():Observable<Array<Class>>
  {
    return this.httpClient.get<Array<Class>>(this.url);
  }
  create(classToCreate:Class) :Observable<Class> {
    debugger;
    const httpOptions = {
      headers: new HttpHeaders({
        'Content-Type': 'application/json',
        'Authorization': 'my-auth-token'
      })
    };
    let _class = classToCreate;
    return this.httpClient.post<Class>(this.url, _class, httpOptions);
  }
  delete(id : number) : Observable<object>
  {
    let deleteUrl = `${this.url}/${id}`;
    return this.httpClient.delete(deleteUrl);
  }
}

File: Backend/Academy.Application/Classes/ClassService.cs
using System.Collections.Generic;
using Academy.Domain.Model.Classes;

namespace Academy.Application.Classes
{
    public class ClassService : IClassService
    {
        private readonly IClassRepository _repository;

        public ClassService(IClassRepository repository)
        {
            this._repository = repository;
        }
        public void Create(Class @class)
        {
            _repository.Add(@class);
        }

        public void Delete(long id)
        {
            var @class = _repository.GetClassById(id);
            if (!ClassIsNull(@class))
                _repository.Delete(@class);
        }

        private bool ClassIsNull(Class @class)
        {
            return @class == null;
        }
        public List<Class> GetAll()
        {
            return _repository.GetAll();
        }
    }
}